### PR TITLE
[3.x] fix cast paginate per page to int

### LINKED ISSUE #1750: Pagination does not accept string number
## Summary

When calling the `->paginate()` method with the first (`perPage`) argument, mongodb errors if the user passes in a string.\

## Steps to reproduce

1. pass `per_page=20` as a query parameter on an application running this library 
2. use `->paginate($request->input('per_page'))` in the called method
3. get a Mongodb error `the limit must be specified as a number`

## Fix

We should cast the `$perPage` argument to an integer before passing to mongodb; this works in the default Laravel drivers, and should work with this driver. This can be easily solved by a simple `(int)` cast

### PR DESCRIPTION
**Pull request by Yahatix**
Cast per_page parameter Collection::paginate to an integer
Added additional test for this case.
Resolves #1750

### CODE REVIEW

File: tests/QueryTest.php [hunk 1]
```diff
@@ -327,6 +327,20 @@ public function testPaginate(): void
         $this->assertEquals(1, $results->currentPage());
     }
 
+    public function testThatPaginateCastsPerPageParameter()
+    {
+        $results = User::paginate('2');
```
Comment (Smolevich) [inline tests/QueryTest.php]: What is the reason pass $perPage as string, if original [method](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Database/Eloquent/Builder.php#L732-L741) expect int or null ?

Comment (Yahatix) [reply tests/QueryTest.php]: See issue: #1750

Comment (Smolevich) [reply tests/QueryTest.php]: Tests in laravel don't have case with pass string `$perPage` https://github.com/laravel/framework/blob/0b12ef19623c40e22eff91a4b48cb13b3b415b25/tests/Integration/Database/QueryBuilderTest.php#L181
https://github.com/laravel/framework/blob/0b12ef19623c40e22eff91a4b48cb13b3b415b25/tests/Database/DatabaseQueryBuilderTest.php#L3205

Comment (Yahatix) [reply tests/QueryTest.php]: So... Whats the problem with adding additional tests?

Comment (Smolevich) [reply tests/QueryTest.php]: I don't understand why we  pass not int value

Comment (Yahatix) [reply tests/QueryTest.php]: Because it's possible when using the default laravel model without mongodb

Comment (Smolevich) [reply tests/QueryTest.php]: i saw in [docs](https://laravel.com/docs/7.x/pagination) and tests and didn't find examples with passing string `$perPage`

Comment (Yahatix) [reply tests/QueryTest.php]: To make you happy, i just tested it in a stock laravel installation and confirmed that it actually does work in a stock installation.
And to no surprise it does.
Granted, when i created the PR i didn't fact check the original issue

**Pull request closed without merging**